data/src/main/java/com/mashup/gabbangzip/sharedalbum/data/repository/GroupRepositoryImpl.kt

```diff
@@ -0,0 +1,31 @@
+package com.mashup.gabbangzip.sharedalbum.data.repository
+
+import com.mashup.gabbangzip.sharedalbum.data.dto.request.CreateGroupRequest
+import com.mashup.gabbangzip.sharedalbum.data.service.GroupService
+import com.mashup.gabbangzip.sharedalbum.domain.model.GroupInfo
+import com.mashup.gabbangzip.sharedalbum.domain.model.GroupParam
+import com.mashup.gabbangzip.sharedalbum.domain.repository.GroupRepository
+import javax.inject.Inject
+
+class GroupRepositoryImpl @Inject constructor(
+    private val groupService: GroupService,
+) : GroupRepository {
+    override suspend fun createGroup(groupParam: GroupParam): GroupInfo {
+        val request = CreateGroupRequest(
+            groupName = groupParam.groupName,
+            keyword = groupParam.keyword,
+            groupImageUrl = groupParam.groupImageUrl,
+        )
+        return groupService.createGroup(request).let { response ->
+            response.data?.run {
+                GroupInfo(
+                    id = id,
+                    groupName = groupName,
+                    keyword = keyword,
+                    groupImageUrl = groupImageUrl,
+                    groupInvitationUrl = groupInvitationUrl,
+                )
+            } ?: throw IllegalStateException("데이터 없음")
+        }
```

Comment:
```suggestion
        return groupService.createGroup(request).data?.let { 
                GroupInfo(
                    id = it.id,
                    groupName = it.groupName,
                    keyword = it.keyword,
                    groupImageUrl = it.groupImageUrl,
                    groupInvitationUrl = it.groupInvitationUrl,
                )
            }
        } ?: throw IllegalStateException("데이터 없음")
```

이렇게 작성하면 뭔가 더 깔끔할거 같은뎅..? 스코프 함수들을 중첩하는게 막막 좋아보이지는 않아서 이렇게 하나로 써도 throw할 수 있을거 같아!

추가로 궁금한게 저렇게 예외에서 사용되는 string들은 따로 관리하는 방법은 없을까? 그냥 저렇게 다 하드 코딩을 해야 하는 걸까..?ㅠ

Reply:
> 이렇게 작성하면 뭔가 더 깔끔할거 같은뎅..? 스코프 함수들을 중첩하는게 막막 좋아보이지는 않아서 이렇게 하나로 써도 throw할 수 있을거 같아!

오 그렇네 이거 못봤네.. 예리해..
Exception String 관리는 다른 코멘트에서 지혜가 추후수정해주는 것으로 확인해서 이번 PR 에서는 패스할게!